homeassistant/components/screenlogic/number.py

```diff
@@ -131,3 +162,55 @@ async def async_set_native_value(self, value: float) -> None:
             ) from sle
         _LOGGER.debug("Set '%s' to %s", self._data_key, value)
         await self._async_refresh()
+
+
+class ScreenLogicSuperChlorinationNumber(ScreenLogicNumber):
+    """Class to represent a ScreenLogic Super Chlorination number."""
+
+    _attr_native_value = 0
+    _attr_native_max_value = 72
+    _attr_native_min_value = 0
+    _attr_native_step = 1
+    _attr_native_unit_of_measurement = get_ha_unit(UNIT.HOUR)
+
+    def __init__(
+        self,
+        coordinator: ScreenlogicDataUpdateCoordinator,
+        entity_description: ScreenLogicNumberDescription,
+    ) -> None:
+        """Initialize a ScreenLogic Super Chlorinate Runtime number entity."""
+        super().__init__(coordinator, entity_description)
+        coordinator.super_chlor_entities[DOMAIN] = self
+
+    @property
+    def entity_data(self) -> dict:
+        """The data for this entity."""
+        return {
+            ATTR.NAME: "Super Chlorination Runtime",
+            ATTR.VALUE: self._attr_native_value,
+        }
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Update the current value."""
+
+        # Current API requires int values for this number.
+        self._attr_native_value = runtime = int(value)
+
+        sc_switch: SwitchEntity | None = self.coordinator.super_chlor_entities.get(
+            SWITCH_DOMAIN
+        )
```

Comment:
Can you refactor this to avoid entities stored in a shared container?

The number domain should not know about the switch domain.  We need to get this information from the library data instead

Reply:
I can do that for the number as the switch's data is valid from the API.

The number on the other hand, is not updated in the API unless the switch is set to on. Can the switch know about the number entity?

Reply:
Two other options I can think of at the moment:
- Store only the number's `int` value on the coordinator.
  - Pro: Not storing the actual entities.
  - Con: One-off value added to the coordinator.
- Expose this as a service only. 
  - Pro: This functionality matches the library CLI functionality.
  - Con: Users will have to set up their own helpers to have a number and switch in the UI. Switch has the potential to get out of sync with pool controller.

I'd really like to avoid having the library manage data that is not directly from the pool controller. It's not currently set up to do so and there are implications in implementing that.

Reply:
We don't want virtual entities so the service is the way to go with the current HA design